indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -245,3 +264,73 @@ def _get_form_defaults(self):
 class RHSessionBlocks(RHManageSessionBase):
     def _process(self):
         return jsonify_template('events/sessions/management/session_blocks.html', sess=self.session)
+
+
+class RHManageSessionTypes(RHManageSessionsBase):
+    """Dialog to manage the session types of an event"""
+
+    def _process(self):
+        types = SessionType.query.with_parent(self.event).all()
+        return jsonify_template('events/sessions/management/types_dialog.html', event=self.event,
+                                types=types)
+
+
+class RHManageSessionTypeBase(RHManageSessionsBase):
+    """Manage a session type of an event"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.session_type
+        }
+    }
+
+    def _process_args(self):
+        RHManageSessionsBase._process_args(self)
+        self.session_type = SessionType.get_one(request.view_args['session_type_id'])
+
+
+class RHEditSessionType(RHManageSessionTypeBase):
+    """Dialog to edit a SessionType"""
+
+    def _process(self):
+        form = SessionTypeForm(event=self.event, obj=self.session_type)
+        if form.validate_on_submit():
+            old_name = self.session_type.name
+            form.populate_obj(self.session_type)
+            db.session.flush()
+            self.event.log(EventLogRealm.management, EventLogKind.change, 'Sessions',
+                           'Updated type: {}'.format(old_name), session.user)
+            return jsonify_data(html_row=render_session_type_row(self.session_type), flash=False)
+        return jsonify_form(form)
+
+
+class RHCreateSessionType(RHManageSessionsBase):
+    """Dialog to add a SessionType"""
+
+    def _get_response(self, new_type):
```

Comment:
no need to split this into a separate method; we only did it in some other places where we wanted to override it in a subclass